File: wdlibrary/src/main/java/com/campanula/library/base/BaseActivity.java
package com.campanula.library.base;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * package ${package_name}
 *
 * @author ${user}
 * create ${date}
 * @since
 **/
public abstract class BaseActivity extends AppCompatActivity {
    private String TAG = BaseActivity.class.getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TAG = tag();
        setRequestWindowFeature();
        setContentView(getLayoutViewId());
        viewById();
        bindData();
    }

    /**
     * 初始化控件数据
     */
    protected abstract void bindData();

    /**
     * 寻找当前页面的View
     */
    protected abstract void viewById();

    /**
     * 获取当前页面layout id
     *
     * @return layout id
     */
    @LayoutRes
    protected abstract int getLayoutViewId();

    /**
     * 在setContentView之前执行
     */
    protected void setRequestWindowFeature() {
    }

    /**
     * 打印日志需要的TAG信息
     *
     * @return Tag
     */
    protected abstract String tag();


}

File: WdLibrary/src/main/java/com/campanula/base/BaseActivity.kt
package com.campanula.base

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    private lateinit var TAG: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        TAG = getPackagePath()
        setRequestWindowFeature()
        setContentView(getLayoutViewId())
        findActivityViewById()
        bindViewData()
    }

    /**
     * 取得Activity的包的路径
     */
    protected abstract fun getPackagePath(): String

    /**
     * 绑定初始View数据
     */
    protected abstract fun bindViewData()

    /**
     * findViewId
     */
    protected abstract fun findActivityViewById()

    /**
     * 设置页面的属性，可以重写，在setContentView之前执行
     */
    protected fun setRequestWindowFeature() {

    }

    @LayoutRes
    protected abstract fun getLayoutViewId(): Int


}